qml/harbour-berlin-vegan.qml

```diff
@@ -62,6 +62,31 @@ ApplicationWindow
         }
     }
 
+    JsonListModel {
+        id: jsonShoppingModel
+        dynamicRoles: true
+    }
+
+    Collection {
+        id: gjsonShoppingModelCollection
+
+        property string searchString: ""
+        property bool   loaded: false
+
+        onSearchStringChanged: reFilter()
+
+        model: jsonShoppingModel
+
+        comparator: function lessThan(a, b) {
```

Comment:
Regarding your question in the commit message: Yes, QML offers inheritance. If, in your example, you add a file
```MyCollection.qml``` and put something in it like
```
Collection {
        property string searchString: ""
        property bool   loaded: false

        onSearchStringChanged: reFilter()

        model: jsonShoppingModel

        comparator: function lessThan(a, b) {
...
```
you would be inheriting from Collection. If putting this in an extra file is too cumbersome for you, you can also use this ```Component``` type to declare a component inline within a file: http://doc.qt.io/qt-5/qml-qtqml-component.html

And yes, answering your second question, that whole collection business is one of the thing that we want to move over to C++ in the mind run.